### PR TITLE
Add all datadog monitors

### PR DESCRIPTION
**Pull request by dannyroberts**
##### SUMMARY
Builds on top of https://github.com/dimagi/commcare-cloud/pull/2771, and uses the tools built out in https://github.com/dimagi/commcare-cloud/pull/2772. After this PR, all our monitors are tracked in code, so we can see in diffs any manual changes, or we can change our notification settings in code and roll them out to all monitors.

##### COMPONENT NAME
Datadog Monitors

##### ADDITIONAL INFORMATION

Diff to be applied (sorry it's long):

```
Diff for 'PostgreSQL replication delay is high'

    ---
    +++
    @@ -3,19 +3,24 @@
       {{/is_alert}}
       {{#is_no_data}}Check that the standby server is running.
       {{/is_no_data}}
    -  {{#is_match "environment.name" "icds-new"}} @slack-alerts-icds  {{/is_match}}
    -  {{#is_match "environment.name" "pna"}} @slack-alerts-pna   {{/is_match}}
    -  {{#is_match "environment.name" "enikshay"}} {{/is_match}}
    -  {{#is_match "environment.name" "india"}} @slack-alerts-softlayer {{/is_match}}
    -  {{#is_match "environment.name" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -  {{^is_match "environment.name" "icds-new"}}
    -      {{^is_match "environment.name" "enikshay"}}
    -          {{^is_match "environment.name" "india"}}
    -              {{^is_match "environment.name" "production"}}
    -                  @slack-hq-ops-priority
    -              {{/is_match}}
    +  {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
    +  {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    +  {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +  {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +  {{^is_match "host.environment" "softlayer"}}
    +    {{^is_match "host.environment" "india"}}
    +      {{^is_match "host.environment" "swiss"}}
    +        {{^is_match "host.environment" "production"}}
    +          {{^is_match "host.environment" "pna"}}
    +            {{^is_match "host.environment" "icds"}}
    +  @slack-hq-ops-priority
    +            {{/is_match}}
               {{/is_match}}
    +        {{/is_match}}
           {{/is_match}}
    +    {{/is_match}}
       {{/is_match}}
     name: PostgreSQL replication delay is high
     options:

Diff for 'ES Cluster Status ALERT'

    ---
    +++
    @@ -1,41 +1,51 @@
     message: |-
       https://confluence.dimagi.com/display/commcarehq/Firefighting+HQ#FirefightingHQ-Elasticsearch
       {{#is_alert}}
    -  {{#is_match "environment" "production"}}
    -  For details check [hqes0](http://hqes0.internal-va.commcarehq.org:9200/_cluster/health?pretty=true), [hqes1](http://hqes1.internal-va.commcarehq.org:9200/_cluster/health?pretty=true) or [hqes3](http://hqes3.internal-va.commcarehq.org:9200/_cluster/health?pretty=true)
    -  {{/is_match}}
    +  For details check http://<es-host>:9200/_cluster/health?pretty=true
       {{/is_alert}}
       {{#is_recovery}}ES cluster status back to GREEN.{{/is_recovery}}
    -  {{#is_match "environment.name" "icds-new"}} @slack-alerts-icds  {{/is_match}}
    -  {{#is_match "environment.name" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -  {{#is_match "environment.name" "softlayer"}} @slack-alerts-softlayer {{/is_match}}
    -  {{#is_match "environment.name" "production"}} @slack-alerts-production {{/is_match}}
    -  {{^is_match "environment.name" "icds-new"}}
    -      {{^is_match "environment.name" "enikshay"}}
    -          {{^is_match "environment.name" "softlayer"}}
    -              {{^is_match "environment.name" "production"}}
    -                  @slack-hq-ops-priority
    -              {{/is_match}}
    +  {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
    +  {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    +  {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +  {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +  {{^is_match "host.environment" "softlayer"}}
    +    {{^is_match "host.environment" "india"}}
    +      {{^is_match "host.environment" "swiss"}}
    +        {{^is_match "host.environment" "production"}}
    +          {{^is_match "host.environment" "pna"}}
    +            {{^is_match "host.environment" "icds"}}
    +  @slack-hq-ops-priority
    +            {{/is_match}}
               {{/is_match}}
    +        {{/is_match}}
           {{/is_match}}
    +    {{/is_match}}
       {{/is_match}}
     name: ES Cluster Status ALERT
     options:
       escalation_message: |-
         https://confluence.dimagi.com/display/commcarehq/Firefighting+HQ#FirefightingHQ-Elasticsearch
         ES cluster status is still not OK and should be looked into ASAP.
    -    {{#is_match "environment.name" "icds-new"}} @slack-alerts-icds  {{/is_match}}
    -    {{#is_match "environment.name" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -    {{#is_match "environment.name" "softlayer"}} @slack-alerts-softlayer {{/is_match}}
    -    {{#is_match "environment.name" "production"}} @slack-alerts-production {{/is_match}}
    -    {{^is_match "environment.name" "icds-new"}}
    -        {{^is_match "environment.name" "enikshay"}}
    -            {{^is_match "environment.name" "softlayer"}}
    -                {{^is_match "environment.name" "production"}}
    -                    @slack-hq-ops-priority
    -                {{/is_match}}
    +    {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +    {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +    {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
    +    {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    +    {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +    {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +    {{^is_match "host.environment" "softlayer"}}
    +      {{^is_match "host.environment" "india"}}
    +        {{^is_match "host.environment" "swiss"}}
    +          {{^is_match "host.environment" "production"}}
    +            {{^is_match "host.environment" "pna"}}
    +              {{^is_match "host.environment" "icds"}}
    +    @slack-hq-ops-priority
    +              {{/is_match}}
                 {{/is_match}}
    +          {{/is_match}}
             {{/is_match}}
    +      {{/is_match}}
         {{/is_match}}
       include_tags: false
       locked: false

Diff for 'Couch Pillow Rewind? Couch Pillow {{pillow_name.name}} on {{environment.name}} is very high'

    ---
    +++
    @@ -1,15 +1,24 @@
     message: |-
       The {{pillow_name.name}} on {{environment.name}} has been above{{#is_warning}} {{warn_threshold}} {{/is_warning}}{{#is_alert}} {{threshold}} {{/is_alert}}for the past minute.
       If this is a pillow rewind, you can solve by finding a couch change seq from before it happened (pick ids from /commcarehq/_changes?descending=true with revs starting with "1-" until you find a form with received_on with an early enough time, and then use the corresonding seq) and use a few lines like https://gist.github.com/dannyroberts/fdd07ecd780e08b6c236092379535a82 in `django-mange shell` to reset the checkpoint to the seq you found.
    -  {{#is_match "environment.name" "icds-new"}} @slack-alerts-icds  {{/is_match}}
    -  {{#is_match "environment.name" "india"}} @slack-alerts-softlayer {{/is_match}}
    -  {{#is_match "environment.name" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -  {{^is_match "environment.name" "icds-new"}}
    -      {{^is_match "environment.name" "softlayer"}}
    -          {{^is_match "environment.name" "production"}}
    -              @slack-hq-ops-priority
    +  {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
    +  {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    +  {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +  {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +  {{^is_match "host.environment" "softlayer"}}
    +    {{^is_match "host.environment" "india"}}
    +      {{^is_match "host.environment" "swiss"}}
    +        {{^is_match "host.environment" "production"}}
    +          {{^is_match "host.environment" "pna"}}
    +            {{^is_match "host.environment" "icds"}}
    +  @slack-hq-ops-priority
    +            {{/is_match}}
               {{/is_match}}
    +        {{/is_match}}
           {{/is_match}}
    +    {{/is_match}}
       {{/is_match}}
     name: Couch Pillow Rewind? Couch Pillow {{pillow_name.name}} on {{environment.name}}
       is very high

Diff for 'Gunicorn 500 errors (as % or requests) {{environment.name}}'

    ---
    +++
    @@ -1,16 +1,25 @@
     message: |-
       {{#is_alert}} Number of 500 is more than 10%. {{/is_alert}}
    -  {{#is_match "environment.name" "icds-new"}} @slack-alerts-icds  {{/is_match}}
    -  {{#is_match "environment.name" "india"}} @slack-alerts-softlayer {{/is_match}}
    +  {{#is_match "environment.name" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "environment.name" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "environment.name" "swiss"}} @slack-alerts-swiss {{/is_match}}
       {{#is_match "environment.name" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -  {{^is_match "environment.name" "icds-new"}}
    -          {{^is_match "environment.name" "india"}}
    -              {{^is_match "environment.name" "production"}}
    -                  @slack-hq-ops-priority
    -              {{/is_match}}
    +  {{#is_match "environment.name" "pna"}} @slack-alerts-pna {{/is_match}}
    +  {{#is_match "environment.name" "icds"}} @slack-alerts-icds {{/is_match}}
    +  {{^is_match "environment.name" "softlayer"}}
    +    {{^is_match "environment.name" "india"}}
    +      {{^is_match "environment.name" "swiss"}}
    +        {{^is_match "environment.name" "production"}}
    +          {{^is_match "environment.name" "pna"}}
    +            {{^is_match "environment.name" "icds"}}
    +  @slack-hq-ops-priority
    +            {{/is_match}}
               {{/is_match}}
    +        {{/is_match}}
    +      {{/is_match}}
    +    {{/is_match}}
       {{/is_match}}
    -name: HQ 500 errors (as % or requests) {{environment.name}}
    +name: Gunicorn 500 errors (as % or requests) {{environment.name}}
     options:
       escalation_message: ''
       include_tags: false

Diff for 'Elasticsearch is returning partial results'

    ---
    +++
    @@ -1,17 +1,23 @@
     message: |-
       Please restart elasticsearch in the short term, and improve cluster stability in the near term (by e.g. adding nodes, resources)
    -  {{#is_match "environment.name" "icds-new"}} @slack-alerts-icds  {{/is_match}}
    -  {{#is_match "environment.name" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -  {{#is_match "environment.name" "india"}} @slack-alerts-softlayer {{/is_match}}
    +  {{#is_match "environment.name" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "environment.name" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "environment.name" "swiss"}} @slack-alerts-swiss {{/is_match}}
       {{#is_match "environment.name" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -  {{^is_match "environment.name" "icds-new"}}
    -      {{^is_match "environment.name" "enikshay"}}
    -          {{^is_match "environment.name" "india"}}
    -              {{^is_match "environment.name" "production"}}
    -                  @slack-hq-ops-priority
    -              {{/is_match}}
    +  {{#is_match "environment.name" "pna"}} @slack-alerts-pna {{/is_match}}
    +  {{#is_match "environment.name" "icds"}} @slack-alerts-icds {{/is_match}}
    +  {{^is_match "environment.name" "softlayer"}}
    +    {{^is_match "environment.name" "india"}}
    +      {{^is_match "environment.name" "swiss"}}
    +        {{^is_match "environment.name" "production"}}
    +          {{^is_match "environment.name" "pna"}}
    +            {{^is_match "environment.name" "icds"}}
    +  @slack-hq-ops-priority
    +            {{/is_match}}
               {{/is_match}}
    +        {{/is_match}}
           {{/is_match}}
    +    {{/is_match}}
       {{/is_match}}
     name: Elasticsearch is returning partial results
     options:

Diff for 'Disk space running low on {{host.name}} ({{device.name}})'

    ---
    +++
    @@ -2,18 +2,24 @@
       {{#is_alert}}ALERT: Disk usage {{comparator}} {{threshold}} ({{value}}) {{/is_alert}}
       {{#is_warning}}WARNING: Disk usage {{comparator}} {{warn_threshold}} ({{value}}) {{/is_warning}}
       https://dimagi.github.io/commcare-cloud/firefighting/#full-drives--out-of-disk-space
    -  {{#is_match "environment.name" "icds"}} @slack-alerts-icds  {{/is_match}}
    -  {{#is_match "environment.name" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -  {{#is_match "environment.name" "india"}} @slack-alerts-softlayer {{/is_match}}
    -  {{#is_match "environment.name" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -  {{^is_match "environment.name" "icds"}}
    -      {{^is_match "environment.name" "enikshay"}}
    -          {{^is_match "environment.name" "india"}}
    -              {{^is_match "environment.name" "production"}}
    -                  @slack-hq-ops-priority
    -              {{/is_match}}
    +  {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
    +  {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    +  {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +  {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +  {{^is_match "host.environment" "softlayer"}}
    +    {{^is_match "host.environment" "india"}}
    +      {{^is_match "host.environment" "swiss"}}
    +        {{^is_match "host.environment" "production"}}
    +          {{^is_match "host.environment" "pna"}}
    +            {{^is_match "host.environment" "icds"}}
    +  @slack-hq-ops-priority
    +            {{/is_match}}
               {{/is_match}}
    +        {{/is_match}}
           {{/is_match}}
    +    {{/is_match}}
       {{/is_match}}
     name: Disk space running low on {{host.name}} ({{device.name}})
     options:

Diff for 'Disk space running low on Riak Node {{host.name}}'

    ---
    +++
    @@ -3,18 +3,24 @@
       {{#is_warning}}Add a riak node in the next month (at the latest) or revisit the cost of comparison to increasing node size.{{/is_warning}}
       {{#is_alert}}Add a riak node in the next 3 days. You missed your window to consider increasing node size.{{/is_alert}}
       (Otherwise it's up to you to figure out what this alert means.)
    -  {{#is_match "environment.name" "icds"}} @slack-alerts-icds  {{/is_match}}
    -  {{#is_match "environment.name" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -  {{#is_match "environment.name" "india"}} @slack-alerts-softlayer {{/is_match}}
    -  {{#is_match "environment.name" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -  {{^is_match "environment.name" "icds"}}
    -      {{^is_match "environment.name" "enikshay"}}
    -          {{^is_match "environment.name" "india"}}
    -              {{^is_match "environment.name" "production"}}
    -                  @slack-hq-ops-priority
    -              {{/is_match}}
    +  {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
    +  {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    +  {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +  {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +  {{^is_match "host.environment" "softlayer"}}
    +    {{^is_match "host.environment" "india"}}
    +      {{^is_match "host.environment" "swiss"}}
    +        {{^is_match "host.environment" "production"}}
    +          {{^is_match "host.environment" "pna"}}
    +            {{^is_match "host.environment" "icds"}}
    +  @slack-hq-ops-priority
    +            {{/is_match}}
               {{/is_match}}
    +        {{/is_match}}
           {{/is_match}}
    +    {{/is_match}}
       {{/is_match}}
     name: Disk space running low on Riak Node {{host.name}}
     options:

Diff for 'Pillow {{pillow_name.name}} on {{environment.name}} is not able to keep up with changes'

    ---
    +++
    @@ -1,18 +1,24 @@
     message: |-
       The {{pillow_name.name}} on {{environment.name}} has been above{{#is_warning}} {{warn_threshold}} {{/is_warning}}{{#is_alert}} {{threshold}} {{/is_alert}}for the past 24 hours.
       This could be the result of an outage or it could imply that more resources are required. Please investigate.
    -  {{#is_match "environment.name" "icds-new"}} @slack-alerts-icds  {{/is_match}}
    -  {{#is_match "environment.name" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -  {{#is_match "environment.name" "india"}} @slack-alerts-softlayer {{/is_match}}
    -  {{#is_match "environment.name" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -  {{^is_match "environment.name" "icds-new"}}
    -      {{^is_match "environment.name" "enikshay"}}
    -          {{^is_match "environment.name" "india"}}
    -              {{^is_match "environment.name" "production"}}
    -                  @slack-hq-ops-priority
    -              {{/is_match}}
    +  {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
    +  {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    +  {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +  {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +  {{^is_match "host.environment" "softlayer"}}
    +    {{^is_match "host.environment" "india"}}
    +      {{^is_match "host.environment" "swiss"}}
    +        {{^is_match "host.environment" "production"}}
    +          {{^is_match "host.environment" "pna"}}
    +            {{^is_match "host.environment" "icds"}}
    +  @slack-hq-ops-priority
    +            {{/is_match}}
               {{/is_match}}
    +        {{/is_match}}
           {{/is_match}}
    +    {{/is_match}}
       {{/is_match}}
     name: Pillow {{pillow_name.name}} on {{environment.name}} is not able to keep up with
       changes

Diff for 'CouchDB node is down on {{instance}}'

    ---
    +++
    @@ -1,17 +1,23 @@
     message: |-
       https://confluence.dimagi.com/display/commcarehq/Firefighting+HQ#FirefightingHQ-Couchnodeisdown
    -  {{#is_match "host.environment" "icds-new"}} @slack-alerts-icds  {{/is_match}}
    -  {{#is_match "host.environment" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -  {{#is_match "host.environment" "india"}} @slack-alerts-softlayer {{/is_match}}
    +  {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
       {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -  {{^is_match "host.environment" "icds-new"}}
    -      {{^is_match "host.environment" "enikshay"}}
    -          {{^is_match "host.environment" "softlayer"}}
    -              {{^is_match "host.environment" "production"}}
    -                  @slack-hq-ops-priority
    -              {{/is_match}}
    +  {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +  {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +  {{^is_match "host.environment" "softlayer"}}
    +    {{^is_match "host.environment" "india"}}
    +      {{^is_match "host.environment" "swiss"}}
    +        {{^is_match "host.environment" "production"}}
    +          {{^is_match "host.environment" "pna"}}
    +            {{^is_match "host.environment" "icds"}}
    +  @slack-hq-ops-priority
    +            {{/is_match}}
               {{/is_match}}
    +        {{/is_match}}
           {{/is_match}}
    +    {{/is_match}}
       {{/is_match}}
     name: CouchDB node is down on {{instance}}
     options:

Diff for 'Riak service on {{host.name}}'

    ---
    +++
    @@ -1,35 +1,47 @@
     message: |-
       {{#is_alert}} Cannot connect to Riak service. {{/is_alert}}
       {{#is_recovery}} Riak connections are back to normal. {{/is_recovery}} @dmiller@dimagi.com
    -  {{#is_match "host.environment" "icds-new"}} @slack-alerts-icds  {{/is_match}}
    -  {{#is_match "host.environment" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -  {{#is_match "host.environment" "india"}} @slack-alerts-softlayer {{/is_match}}
    +  {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +  {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
       {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -  {{^is_match "host.environment" "icds-new"}}
    -      {{^is_match "host.environment" "enikshay"}}
    -          {{^is_match "host.environment" "india"}}
    -              {{^is_match "host.environment" "production"}}
    -                  @slack-hq-ops-priority
    -              {{/is_match}}
    +  {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +  {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +  {{^is_match "host.environment" "softlayer"}}
    +    {{^is_match "host.environment" "india"}}
    +      {{^is_match "host.environment" "swiss"}}
    +        {{^is_match "host.environment" "production"}}
    +          {{^is_match "host.environment" "pna"}}
    +            {{^is_match "host.environment" "icds"}}
    +  @slack-hq-ops-priority
    +            {{/is_match}}
               {{/is_match}}
    +        {{/is_match}}
           {{/is_match}}
    +    {{/is_match}}
       {{/is_match}}
     name: Riak service on {{host.name}}
     options:
       escalation_message: |-
         Still cannot connect to Riak service.
    -    {{#is_match "host.environment" "icds-new"}} @slack-alerts-icds  {{/is_match}}
    -    {{#is_match "host.environment" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -    {{#is_match "host.environment" "india"}} @slack-alerts-softlayer {{/is_match}}
    +    {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +    {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +    {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
         {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -    {{^is_match "host.environment" "icds-new"}}
    -        {{^is_match "host.environment" "enikshay"}}
    -            {{^is_match "host.environment" "india"}}
    -                {{^is_match "host.environment" "production"}}
    -                    @slack-hq-ops-priority
    -                {{/is_match}}
    +    {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +    {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +    {{^is_match "host.environment" "softlayer"}}
    +      {{^is_match "host.environment" "india"}}
    +        {{^is_match "host.environment" "swiss"}}
    +          {{^is_match "host.environment" "production"}}
    +            {{^is_match "host.environment" "pna"}}
    +              {{^is_match "host.environment" "icds"}}
    +    @slack-hq-ops-priority
    +              {{/is_match}}
                 {{/is_match}}
    +          {{/is_match}}
             {{/is_match}}
    +      {{/is_match}}
         {{/is_match}}
       include_tags: false
       locked: false

Diff for '[ICDS] Disk space will reach {{threshold}}% in 2 weeks'

    ---
    +++
    @@ -5,18 +5,24 @@
     options:
       escalation_message: |-
         Disk Space Used will cross the  {{threshold}} mark in next 2 weeks.
    -    {{#is_match "environment.name" "icds"}} @slack-alerts-icds  {{/is_match}}
    -    {{#is_match "environment.name" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -    {{#is_match "environment.name" "india"}} @slack-alerts-softlayer {{/is_match}}
    -    {{#is_match "environment.name" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -    {{^is_match "environment.name" "icds"}}
    -        {{^is_match "environment.name" "enikshay"}}
    -            {{^is_match "environment.name" "india"}}
    -                {{^is_match "environment.name" "production"}}
    -                    @slack-hq-ops-priority
    -                {{/is_match}}
    +    {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +    {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +    {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
    +    {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    +    {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +    {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +    {{^is_match "host.environment" "softlayer"}}
    +      {{^is_match "host.environment" "india"}}
    +        {{^is_match "host.environment" "swiss"}}
    +          {{^is_match "host.environment" "production"}}
    +            {{^is_match "host.environment" "pna"}}
    +              {{^is_match "host.environment" "icds"}}
    +    @slack-hq-ops-priority
    +              {{/is_match}}
                 {{/is_match}}
    +          {{/is_match}}
             {{/is_match}}
    +      {{/is_match}}
         {{/is_match}}
       include_tags: true
       locked: false

Diff for '[Scale] Disk space will reach {{threshold}} on {{host.name}} in 2 weeks'

    ---
    +++
    @@ -8,18 +8,24 @@
     options:
       escalation_message: |-
         Disk Space Used will cross the  {{threshold}} mark in next 2 weeks.
    -    {{#is_match "environment.name" "icds"}} @slack-alerts-icds  {{/is_match}}
    -    {{#is_match "environment.name" "enikshay"}} @slack-alerts-enikshay {{/is_match}}
    -    {{#is_match "environment.name" "india"}} @slack-alerts-softlayer {{/is_match}}
    -    {{#is_match "environment.name" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    -    {{^is_match "environment.name" "icds"}}
    -        {{^is_match "environment.name" "enikshay"}}
    -            {{^is_match "environment.name" "india"}}
    -                {{^is_match "environment.name" "production"}}
    -                    @slack-hq-ops-priority
    -                {{/is_match}}
    +    {{#is_match "host.environment" "softlayer"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +    {{#is_match "host.environment" "india"}} @slack-alerts-softlayer @slack-alerts-india {{/is_match}}
    +    {{#is_match "host.environment" "swiss"}} @slack-alerts-swiss {{/is_match}}
    +    {{#is_match "host.environment" "production"}} @slack-alerts-production @opsgenie-commcarehq {{/is_match}}
    +    {{#is_match "host.environment" "pna"}} @slack-alerts-pna {{/is_match}}
    +    {{#is_match "host.environment" "icds"}} @slack-alerts-icds {{/is_match}}
    +    {{^is_match "host.environment" "softlayer"}}
    +      {{^is_match "host.environment" "india"}}
    +        {{^is_match "host.environment" "swiss"}}
    +          {{^is_match "host.environment" "production"}}
    +            {{^is_match "host.environment" "pna"}}
    +              {{^is_match "host.environment" "icds"}}
    +    @slack-hq-ops-priority
    +              {{/is_match}}
                 {{/is_match}}
    +          {{/is_match}}
             {{/is_match}}
    +      {{/is_match}}
         {{/is_match}}
       include_tags: true
       locked: false
```

### CODE REVIEW

**Comment by millerdev** [comment]
That is a lot of code. Wonder if it would make sense to define a default set of properties and then only define deviations from that in these monitor definitions? Merge the two to get the final result.

Also, is there a test or maybe an easy automated way to make sure changes made directly in the datadog web UI get copied back into these definitions?

**Comment by millerdev** [comment]
Looks like https://github.com/dimagi/commcare-cloud/pull/2772 answers my question

**Comment by dannyroberts** [comment]
> That is a lot of code. Wonder if it would make sense to define a default set of properties and then only define deviations from that in these monitor definitions? Merge the two to get the final result.

I think the code Simon wrote already does that to some extent. There's basically a white list of keys that we will update https://github.com/dimagi/commcare-cloud/blob/62666287a15a494896df4a2ed96d7e24cda952dd/src/commcare_cloud/manage_commcare_cloud/datadog_monitors.py#L25-L35 and all other differences are ignored.

**Comment by dannyroberts** [comment]
I like Simon's suggestion of a whitelist; that's a simple way to cut down on files here we don't need

**Comment by dannyroberts** [comment]
FYI I removed the support monitors here, but because of the code involved, I put the code for ignoring specific monitors (and listing these ones) in https://github.com/dimagi/commcare-cloud/pull/2772.

**Comment by dannyroberts** [comment]
Updated the diff in the description FWIW

File: src/commcare_cloud/manage_commcare_cloud/datadog_monitors.py [hunk 1]
```diff
@@ -195,13 +200,19 @@ def __init__(self, errors):
 
 class RemoteMonitorAPI(object):
     def _wrap(self, raw_mon):
+        # The drove me crazy to figure out, but the get_all endpoint omits
+        # options.synthetics_check_id for no reason I can think of
+        # o if it's a synthetics alert, pull it again directly by id
```
Comment (millerdev) [inline src/commcare_cloud/manage_commcare_cloud/datadog_monitors.py]: typo

**Pull request merged**